### ISSUE TITLE
Error while setting up caldav platform for calendar

### ISSUE THREAD

**Issue by paulcombuse** [home-assistant/home-assistant.github.io#26296]
### Feedback

I would like to use the CalDAV integration to connect to my GMX calendars. I can establish a connection on Android using DAVx5 - so basically I assume that the server is compatible. In my configuration YAML the following lines were added by me:
```YAML
calendar:
  - platform: caldav
    url: "https://caldav.gmx.net/MY_EMAIL_ADDRESS/"
    username: !secret userGMX
    password: !secret passGMX
```
And after a restart of HomeAssistant I can see the two calendars "Mein Kalender" and "Geburtstage" - meaning that some kind of connection was successfully established. Though, there are no items in the calendars and an error message says: "Events for this calendar could not be retrieved: Mein Kalender".

The LOGs of HA core show the following error:
```YAML
Logger: homeassistant.components.calendar
Source: components/caldav/calendar.py:87
Integration: Kalender (documentation, issues)
First occurred: 19:29:01 (1 occurrences)
Last logged: 19:29:01

Error while setting up caldav platform for calendar
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 293, in _async_setup_platform
    await asyncio.shield(task)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/caldav/calendar.py", line 87, in setup_platform
    calendars = client.principal().calendars()
  File "/usr/local/lib/python3.10/site-packages/caldav/davclient.py", line 412, in principal
    self._principal = Principal(client=self, *largs, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/caldav/objects.py", line 478, in __init__
    cup = self.get_property(dav.CurrentUserPrincipal())
  File "/usr/local/lib/python3.10/site-packages/caldav/objects.py", line 210, in get_property
    foo = self.get_properties([prop], **passthrough)
  File "/usr/local/lib/python3.10/site-packages/caldav/objects.py", line 242, in get_properties
    properties = response.expand_simple_props(props)
  File "/usr/local/lib/python3.10/site-packages/caldav/davclient.py", line 293, in expand_simple_props
    self.find_objects_and_props()
  File "/usr/local/lib/python3.10/site-packages/caldav/davclient.py", line 216, in find_objects_and_props
    responses = self._strip_to_multistatus()
  File "/usr/local/lib/python3.10/site-packages/caldav/davclient.py", line 152, in _strip_to_multistatus
    if tree.tag == "xml" and tree[0].tag == dav.MultiStatus.tag:
AttributeError: 'NoneType' object has no attribute 'tag'
```
The funny thing is that for the very last line of this error the corresponding HTML page can is [https://caldav.gmx.net/](https://caldav.gmx.net/) - meaning it seems that the URL is somewhat modified and the calendars are therefore not reachable.

I called the page "https://caldav.gmx.net/MY_EMAIL_ADDRESS" from where I am redirected to "https://caldav.gmx.net/begenda/dav/users/MY_EMAIL_ADDRESS". After logging in with the above stated _userGMX_ and _passGMX_ I see an HTML page with the following content:
```HTML
MY_NAME
Properties
{DAV:}creationdate
2023-02-17T18:25:04Z
{DAV:}getetag
klasdhflkahsdlkhföaskdhihaäpireh=
{urn:ietf:params:xml:ns:caldav}calendar-home-set
/begenda/dav/a12bc3d4-56ef-7890-1234-5ghi67jk8901
{DAV:}getlastmodified
Fri, 17 Feb 2023 18:25:04 GMT
{urn:ietf:params:xml:ns:caldav}schedule-outbox-URL
/begenda/dav/a12bc3d4-56ef-7890-1234-5ghi67jk8901/Outbox
{DAV:}iscollection
0
{DAV:}supported-report-set
{DAV:}principal-match
{DAV:}acl
org.unitedinternet.cosmo.dav.acl.DavAcl@1dkh390klsl
{urn:ietf:params:xml:ns:caldav}calendar-user-address-set
MY_EMAIL_ADDRESS
{DAV:}resourcetype
{DAV:}principal
{urn:ietf:params:xml:ns:caldav}schedule-inbox-URL
/begenda/dav/a12bc3d4-56ef-7890-1234-5ghi67jk8901/Inbox
{DAV:}current-user-privilege-set
{DAV:}read
{DAV:}alternate-URI-set
{DAV:}displayname
MY_NAME
{DAV:}principal-URL
/begenda/dav/users/a12bc3d4-56ef-7890-1234-5ghi67jk8901
{DAV:}group-membership
[User Principals](https://caldav.gmx.net/begenda/dav/users/)
[Home collection](https://caldav.gmx.net/begenda/dav/a12bc3d4-56ef-7890-1234-5ghi67jk8901/)
```
So basically the address seems to be alright and there seems to be some kind of DAV content, but unfortunately I am doing something wrong or there is some kind of bug. I would highly appreciate help on this.

### URL

https://www.home-assistant.io/integrations/caldav/

### Version

2023.2.5

**Comment by DrDresche** [comment]
I have exactly the same issue, also with GMX.

**Comment by klaasnicolaas** [comment]
Hi there @paulcombuse 👋

It seems like you want to report an issue with an Home Assistant integration, however, you have submitted documentation feedback instead. Please raise an issue in the Home Assistant core issue tracker instead:

https://github.com/home-assistant/core/issues

Thanks! 👍

./Klaas

**Comment by paulcombuse** [comment]
Thanks @klaasnicolaas - I followed your advice and reposted the issue [here](https://github.com/home-assistant/core/issues/90300#issue-1640893288)
